### PR TITLE
[20.09] nextcloud: init at 21.0.0

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

Partial backport of #113958.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/nextcloud.nix [hunk 1]
```diff
@@ -362,9 +386,10 @@ in {
           support upgrades that skip multiple versions (i.e. an upgrade from 17 to 19 isn't
           possible, but an upgrade from 18 to 19).
         '')
-        ++ (optional (versionOlder cfg.package.version "20") ''
+        ++ (optional (versionOlder cfg.package.version "21") ''
           The latest Nextcloud release is v20 which can be installed by setting
```
Comment (berbiche) [inline nixos/modules/services/web-apps/nextcloud.nix:390]:
```suggestion
          The latest Nextcloud release is v21 which can be installed by setting
```

File: nixos/modules/services/web-apps/nextcloud.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
  </para>
  <para>
   The current default by NixOS is <package>nextcloud19</package> though it's recommended
-  to upgrade to the latest version, <package>nextcloud20</package>.
+  to upgrade to the latest version, <package>nextcloud21</package>.
```
Comment (berbiche) [inline nixos/modules/services/web-apps/nextcloud.xml:15]: Maybe the warning about updating to v21 from v19 should be added here?

File: nixos/modules/services/web-apps/nextcloud.nix (hunk 1 above)
Comment (Ma27) [reply nixos/modules/services/web-apps/nextcloud.nix:390]: Good catch!

**Comment by Ma27** [comment]
@berbiche fixed :)

**Pull request merged**